pkg/multiplexer/multiplexer.go

```diff
@@ -130,7 +148,13 @@ func (r *Request) Do(ctx context.Context) ([]*Result, error) {
 	case <-done:
 		return results, nil
 	case <-ctx.Done():
-		return make([]*Result, 0), newTimeoutError(ctx.Err())
+		if errors.Is(ctx.Err(), context.Canceled) {
+			return make([]*Result, 0), newCancellationError(ctx.Err())
+		} else if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+			return make([]*Result, 0), newTimeoutError(ctx.Err())
+		} else {
+			return make([]*Result, 0), ctx.Err()
+		}
```

Comment:
Minor, thoughts on updating this to use a `switch` statement? like:

```
switch {
case errors.Is(ctx.Err(), context.Canceled):
case errors.Is(ctx.Err(), context.DeadlineExceeded):
default:
```

Reply:
I like it! 4fe5e294cd2f1f09e00b72579315bd918c44d66b